diff --git a/swig/device.swg b/swig/device.swg
index 2f2d76b..265ecd6 100644
--- a/swig/device.swg
+++ b/swig/device.swg
@@ -36,11 +36,11 @@ typedef char ** DeviceList_t;
       /* For each element in the array of strings, push that
        * onto the result object. */
       for ( ; *p; p++) {
-#if PY_VERSION_HEX >= 0x03000000
+%#if PY_VERSION_HEX >= 0x03000000
 	PyObject *o = PyUnicode_FromStringAndSize(*p, strlen(*p));
-#else
+%#else
 	PyObject *o = PyString_FromStringAndSize(*p, strlen(*p));
-#endif
+%#endif
 	PyList_Append(resultobj, o);
       }
     }
